src/main/java/pl/pandziak/ticketbookingapp/domain/reservation/ReservationInMemoryRepository.java

```diff
@@ -0,0 +1,41 @@
+package pl.pandziak.ticketbookingapp.domain.reservation;
+
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+final class ReservationInMemoryRepository implements ReservationRepository {
+
+    private final Map<String, Reservation> reservations = new ConcurrentHashMap<>();
+
+    @Override
+    public Reservation save(Reservation reservation) {
+        reservations.put(reservation.getId(), reservation);
+        return reservation;
+    }
+
+    @Override
+    public Optional<Reservation> findById(String id) {
+        return Optional.ofNullable(reservations.get(id));
+    }
+
+    @Override
+    public List<Reservation> findAll() {
+        return List.copyOf(reservations.values());
```

Comment:
Można by chyba zwrócić coś niemodyfikowalnego, nie?

Reply:
Wydaje mi się, że copyOf zwraca właśnie niemodyfikowalną listę. Początek javadoc: "Returns an unmodifiable List containing the elements of the given Collection, in its iteration order."

Reply:
A, to przepraszam!